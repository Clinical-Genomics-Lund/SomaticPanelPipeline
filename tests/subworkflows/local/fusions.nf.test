nextflow_workflow {

    name "Test Workflow FUSIONS"
    script "subworkflows/local/fusions.nf"
    workflow "FUSIONS"

    setup {
        
        run("CHECK_INPUT") {
            script "$projectDir/subworkflows/local/create_meta.nf"
            workflow {
                """
                input[0] = file("$projectDir/tests-data/gmslymphomav3-0/23KF00457-lymf_gmslymphomav3-0.csv")
                """
            }
        }

        run("SAMPLE") {
            script "$projectDir/subworkflows/local/sample.nf"
            workflow {
                """
                input[0] = CHECK_INPUT.out.fastq
                """
            }
        }

        run("ALIGN_SENTIEON") {
            script "$projectDir/subworkflows/local/align_sentieon.nf"
            workflow {
                """
                input[0] =  SAMPLE.out.fastq_trim
                input[1] =  CHECK_INPUT.out.meta
                """
            }
        }
    }

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            workflow {
                """
                // define inputs of the workflow here. Example:
                input[0] = SAMPLE.out.fastq_trim
                input[1] = CHECK_INPUT.out.meta
                input[2] = ALIGN_SENTIEON.out.bam_dedup
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
