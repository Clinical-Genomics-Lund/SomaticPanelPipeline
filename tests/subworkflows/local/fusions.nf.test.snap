{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "fusions": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "timestamp": "2023-12-28T15:36:37.027983"
    }
}